src/hstr.c

```diff
@@ -1015,31 +1053,41 @@ void loop_to_select(Hstr *hstr)
     recalculate_max_history_items();
 
     // print prompt before pattern
-    basex=print_prompt();
+    print_prompt(hstr);
 
     // print current pattern
-    print_pattern(pattern, hstr->promptY, basex);
+    print_pattern(hstr, pattern);
     cursorX=getcurx(stdscr);
     cursorY=getcury(stdscr);
 
     // initialize history list
-    result=hstr_make_and_print_selection(pattern, hstr);
+    hstr_make_and_print_selection(pattern, hstr);
     color_attr_off(COLOR_PAIR(HH_COLOR_NORMAL));
 
     // initialize info rows
-    print_status_label();
+    print_status_label(hstr);
     print_help_label(); // help is static and almost never needs updating
 
-    // finally, put cursor back at the prompt
-    move(cursorY, cursorX);
-
     while (!done) {
+        if (patternChanged) {
+            patternChanged = false;
+            if (hstr->selectionSize > 0) {
+                selectionCursorPosition=0;
```

Comment:
After using this I think highlight selection should include the matching part of the pattern too.